### PR TITLE
frontend/#5460/Driver Interoperability Feature

### PR DESCRIPTION
**Pull request by ashishsingh101**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

1. Driver Interoperability flow

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added integration tests for my changes where possible
- [ ] I tested the changes using integration tests
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/android-native/mobility-customer/src/main/res/values-hi/strings.xml [hunk 1]
```diff
@@ -50,4 +50,6 @@
     <string name="driver_receipt_downloaded">Driver Receipt Downloaded!!!</string>
     <string name="driver_receipt_for_your_ride_is_downloaded">Driver Receipt for your ride is downloaded!!!</string>
     <string name="invoice_for_your_ride_is_downloaded">Invoice for your ride is downloaded!!!</string>
+    <string name="no_additional_fare_tips_and_waiting_charges">"कोई अतिरिक्त किराया नहीं। कोई सुझाव नहीं। कोई प्रतीक्षा शुल्क नहीं।"</string>
+    <string name="third_party_booking">"तीसरे पक्ष की बुकिंग"</string>
```
Comment (Rohit4997) [inline Frontend/android-native/mobility-customer/src/main/res/values-hi/strings.xml:54]: These strings are not used in mobility customer, Please remove from mobility-customer

**Review by Rohit4997 — CHANGES_REQUESTED** [review]

File: Frontend/ui-driver/src/Screens/HomeScreen/View.purs [hunk 2]
```diff
@@ -288,6 +288,7 @@ view push state =
       ][ Anim.screenAnimationFadeInOut $
           driverMapsHeaderView push state
         , rideActionModelView push state
+        -- , if state.data.activeRide.bookingFromOtherPlatform then RideActionModal.bottomPlatformInfoBar VISIBLE else dummyTextView
```
[the remark this answers is not in the document]
Comment (ashishsingh101) [reply Frontend/ui-driver/src/Screens/HomeScreen/View.purs:291]: need this

File: Frontend/ui-driver/src/Components/RideActionModal/View.purs [hunk 3]
```diff
@@ -1229,4 +1237,36 @@ stopImageView  config push =
             , margin $ MarginTop 4
             ],
             SeparatorView.view separatorConfig]
-    ]
\ No newline at end of file
+    ]
+    
+bottomPlatformInfoBar :: forall w. Visibility -> PrestoDOM (Effect Unit) w
```
[the remark this answers is not in the document]
Comment (ashishsingh101) [reply Frontend/ui-driver/src/Components/RideActionModal/View.purs:1242]: commented as we may need this in future

File: Frontend/android-native/app/src/main/res/values-hi/strings.xml [hunk 4]
```diff
@@ -48,4 +48,6 @@
     <string name="message_from_customer">ग्राहक की ओर से संदेश </string>
     <string name="go_to">गो-टू</string>
     <string name="call_support">सहायता कॉल करें</string>
+    <string name="no_additional_fare_tips_and_waiting_charges">"कोई अतिरिक्त किराया नहीं। कोई सुझाव नहीं। कोई प्रतीक्षा शुल्क नहीं।"</string>
+    <string name="third_party_booking">"तीसरे पक्ष की बुकिंग"</string>
```
[the remark this answers is not in the document]
Comment (ashishsingh101) [reply Frontend/android-native/app/src/main/res/values-hi/strings.xml:52]: online articles prefer `तृतीय` so replaced with that

File: Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/OverlaySheetService.java [hunk 5]
```diff
@@ -778,7 +779,10 @@ private void showOverLayPopup(Bundle rideRequestBundle) {
         }catch (Exception e){
             e.printStackTrace();
         }
-        if (key != null && key.equals("yatrisathiprovider")){
+        if (isThirdPartyBooking) {
```
[the remark this answers is not in the document]
Comment (ashishsingh101) [reply Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/OverlaySheetService.java:782]: as per new design will have to show `nammaYatri` tag so removed this logic only

File: Frontend/ui-driver/src/Resource/Localizable/EN.purs [hunk 6]
```diff
@@ -1413,3 +1413,18 @@ getEN stringKey =
...
         ADVANCED_RIDE_POPUP_TITLE -> "Get rides even before your current ride ends!"
+        THIRD_PARTY_BOOKING -> "Third party booking"
+        SOME_FEATURE_ARE_NOT_AVAILABLE_WITH_THIS_PROVIDER merchant -> "Some " <> merchant <> " Features are not available with this provider."
+        GUARANTEED_RIDE -> "Guaranteed Ride"
+        CUSTOMER_CALLING_AND_MESSAGING -> "Customer Calling & Messaging"
+        WAITING_CHARGES -> "Waiting Charges"
+        CUSTOMER_TIPS -> "Customer Tips"
+        CANCELLATION_CHARGES -> "Cancellation Charges"
+        MERCHANT_COINS merchant -> merchant <> " Coins"
+        MERCHANT_NAME _ -> "Namma Yatri"
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Resource/Localizable/EN.purs:1424]: use merchant name from config

**Review by MercyQueen — CHANGES_REQUESTED** [review]

**Review by MercyQueen — DISMISSED** [review]

**Review by aryan-13 — APPROVED** [review]

**Review by Rohit4997 — APPROVED** [review]

**Pull request merged**